### ISSUE TITLE
Regression in 1.9.4 in HTML Tree View (command \mainpage)

### ISSUE THREAD

**Issue by Albrecht-S** [doxygen/doxygen#9616]
**Describe the bug**
Beginning with Doxygen 1.9.4 the HTML tree view does no longer show subpages of the main page created with `\mainpage` in the left "treeview" panel. Examples and `git bisect` results below.

**Expected behavior**
Doxygen 1.9.4 and later should produce the same output as previous versions.

**Screenshots**

Good: Doxygen 1.9.1
![doxygen_1 9 1](https://user-images.githubusercontent.com/26235500/192488271-bed931db-9c68-42b9-b79f-3e3c64e9dc38.png)

Bad: Doxygen 1.9.4 (missing subpages)
![doxygen_1 9 4](https://user-images.githubusercontent.com/26235500/192488309-06ddf7df-c277-4b41-bbcf-ddf602f58d78.png)

**To Reproduce**
See attached file [doxygen_mainpage.tar.gz](https://github.com/doxygen/doxygen/files/9654243/doxygen_mainpage.tar.gz)
- Doxyfile: created with 1.9.4, edited: `GENERATE_TREEVIEW = YES` and `GENERATE_LATEX = NO`
- index.cxx: the only source file

**Version**
Doxygen 1.9.4 (self-built from tag Release_1_9_4): fails
Doxygen 1.9.1 (self-built from tag Release_1_9_1): OK

Doxygen commit 968e10e90f769d86423ef0fe2f2867038ccdd248: fails (see bisect output below)
Previous commit 15dae504a167d838873053cc96417d8beae78df7: OK

Doxygen master (36b1fa7835251a23132b90f76b557ef3e83d31a1): fails

System: Linux Mint 20 (based on Ubuntu 20.04)
```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Linuxmint
Description:	Linux Mint 20
Release:	20
Codename:	ulyana
```

**Additional context**
`git bisect` shows that this commit introduces the regression:
```
$ git bisect bad
968e10e90f769d86423ef0fe2f2867038ccdd248 is the first bad commit
commit 968e10e90f769d86423ef0fe2f2867038ccdd248
Author: albert-github <albert.tests@gmail.com>
Date:   Mon Feb 21 12:44:28 2022 +0100

    Incorrect representation of some items in HTMLHELP
    
    When trying to generate the doxygen internal documentation with htmlhelp we see a number of items that are shown as pages but actually have sub pages and should thus be shown as books.
    - general documentation
    - namespace members
    - file members

 src/index.cpp | 24 +++++++++++++++---------
 1 file changed, 15 insertions(+), 9 deletions(-)
```

This problem was initially reported for the [FLTK project](https://www.fltk.org/) in https://github.com/fltk/fltk/issues/505. This [comment](https://github.com/fltk/fltk/issues/505#issuecomment-1256957815) contains more screenshots of the docs created by Doxygen 1.9.1 and 1.9.4, respectively.

The example project posted above is a massively stripped down project taken from the original project. The full documentation of FLTK can be found on our [documentation page](https://www.fltk.org/doc-1.4/index.html).

It is important in this context that the main page `index.html` shows the full index page in the right panel and that the treeview in the left panel shows the subpages (menu items) not only in the first view (`index.html`) but also when navigating through the pages.

A proposed workaround to change `\mainpage ...` to `\page ...` does not produce the desired result.

Thanks for the great Doxygen project!

**Comment by albert-github** [comment]
An initial guess is that the true / false setting etc. have the be done depending on the setting of `GENERATE_TREEVIEW`. The `GENERATE_TREEVIEW` is automatically set to `NO` for the HTMLHELP output.

**Comment by Albrecht-S** [comment]
@albert-github Thanks for the quick response and classification as "bug". I take it that means there are chances that it will be fixed soon. ;-)
TIA.

**Comment by doxygen** [comment]
@albert-github If I look at commit https://github.com/doxygen/doxygen/commit/968e10e90f769d86423ef0fe2f2867038ccdd248 it seems like the extra `incContentDepth` and `decContentDepth` in `writeIndex` lines 4354 and 4363 of `index.cpp` are causing the problem. Any idea why these were added, or what breaks if these are removed?

**Comment by albert-github** [comment]
I didn't do tests yet, but the first guess is that seen the description of the PR https://github.com/doxygen/doxygen/pull/9137 that in the HTMLHELP (and probably also QHP)  there will be, al least, incorrect icons.

**Comment by doxygen** [comment]
@Albrecht-S Please verify if the referenced commit fixes the problem.
@albert-github Can you check if the icons are still correct? (as far as I can see they are, but maybe you had a different test case).

**Comment by albert-github** [comment]
I did test with HTMHELP and with QHP and all looks OK to me.

**Comment by Albrecht-S** [comment]
@doxygen @albert-github Thanks for looking into this issue and for the fix. I can confirm that commit c84ec66d1255f8802b35e64c804c57fd67844bdc fixes the issue in my test case and in the original FLTK documentation project.

Since this commit fixes the issue I believe this issue can be closed. Thank you very much.

**Comment by Albrecht-S** [comment]
However, I found a new **_unrelated issue_** that exists already in the previous commit 59aa82c1385054dadfb636375e7e45be6b95672c, i.e. it's not a regression caused by this particular fix.

I see new bogus warnings that are not directly related to our source code:
```
.../preface.dox:74: warning: Unexpected token TK_HTMLTAG found as part of a title section
.../preface.dox:74: warning: Unexpected token TK_HTMLTAG found as part of a title section
.../preface.dox:80: warning: Unexpected token TK_HTMLTAG found as part of a title section
.../preface.dox:80: warning: Unexpected token TK_HTMLTAG found as part of a title section
.../preface.dox:80: warning: Unexpected token TK_HTMLTAG found as part of a title section
.../preface.dox:80: warning: Unexpected token TK_HTMLTAG found as part of a title section
```
Obviously there is no "token `TK_HTMLTAG`" in our source file, neither in line 74 nor in line 80, as you can see [in the original file](https://github.com/fltk/fltk/blob/f2532964cc8b326d00354f11d551727795a99319/documentation/src/preface.dox#L74).

I just wanted to mention this here. I don't think that I'm able to create a reproducer since this is not directly related to our source code (where does this `TK_HTMLTAG` token come from?) . Anyway, I can create another issue if you need one.

**Comment by albert-github** [comment]
Yes please create a new issue for the new problem.

Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by Albrecht-S** [comment]
OK, I'll create a new issue.

**Comment by Albrecht-S** [comment]
FTR: I filed issue #9628: `Bogus warning: "Unexpected token TK_HTMLTAG found as part of a title section"`.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.6.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by Albrecht-S** [comment]
Thanks to all for fixing this issue.

- I confirm that it is **fixed in 1.9.6** (built from source, tag: `Release_1_9_6`).
- I also confirm that it was **not fixed in 1.9.5** (tag: `Release_1_9_5`).
